WPFSKillTree/Utils/Extensions/NumberExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace POESKillTree.Utils.Extensions
+{
+    public static class NumberExtensions
+    {
+        public static bool AlmostEquals(this float x, float y)
+        {
+            // float has 6 significant digits. 4 leaves a bit of room for accumulated errors.
+            var epsilon = Math.Max(Math.Abs(x), Math.Abs(y)) * 1e-5F;
+            return AlmostEquals(x, y, epsilon);
+        }
+        public static bool AlmostEquals(this float x, float y, float epsilon)
+        {
+            return Math.Abs(x - y) <= epsilon;
+        }
+
+        public static bool AlmostEquals(this double x, double y)
+        {
+            // double has 15 significant digits. 13 leaves a bit of room for accumulated errors.
+            var epsilon = Math.Max(Math.Abs(x), Math.Abs(y)) * 1e-14;
+            return AlmostEquals(x, y, epsilon);
```

Comment:
This (and also the float one above) won't work properly if one value is e.g. zero and the other is a denorm (epsilon will be zero and they will therefore be considered unequal).

You're essentially trying to figure out how many ULPs the smaller number is away from the larger one. But ULP-based comparison (as demonstrated) breaks down near zero. [This article](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/) demonstrates it quite well. Since you're actually going through the trouble of writing this extension, I'd probably recommend [this](http://floating-point-gui.de/errors/comparison/) approach.